src/ui/generic-components/TrafficLightButton.swift

```diff
@@ -1,219 +1,379 @@
-        if (type == .fullscreen) {
-            window_?.toggleFullscreen()
-        } else if (type == .miniaturize) {
-            window_?.minDemin()
-        } else if (type == .close) {
-            window_?.close()
-        } else if (type == .quit) {
-            window_?.application.quit()
-        }
-    }
-
+    /// Mouse entered event handler.
     override func mouseEntered(with event: NSEvent) {
         isMouseOver = true
-        setNeedsDisplay()
+        setNeedsDisplay(bounds)
     }
 
+    /// Mouse exited event handler.
     override func mouseExited(with event: NSEvent) {
         isMouseOver = false
-        setNeedsDisplay()
+        setNeedsDisplay(bounds)
+    }
+
+    /// Mouse up event handler.
+    override func mouseUp(with event: NSEvent) {
+        if event.clickCount >= 1 {
+            performClick()
+        }
+        setNeedsDisplay(bounds)
+    }
+
+    /// Perform the button click action.
+    private func performClick() {
+        if let window = targetWindow, let type = type {
+            switch type {
+            case .fullscreen:
+                window.toggleFullscreen()
+            case .miniaturize:
+                window.minDemin()
+            case .close:
+                window.close()
+            case .quit:
+                window.application.quit()
+            }
+        }
     }
 
+    /// Custom drawing of the button.
+    ///
+    /// - Parameters:
+    ///   - dirtyRect: The portion of the view’s bounds that needs to be redrawn.
     override func draw(_ dirtyRect: NSRect) {
+        super.draw(dirtyRect)
+        // Retrieve the colors for the button based on its type and system control tint
         let (diskBackgroundColor, diskStrokeColor, symbolColor) = colors()
-        let disk = drawDisk(diskBackgroundColor, diskStrokeColor)
+
+        let disk = NSBezierPath()
+        disk.appendOval(in: NSMakeRect(bounds.origin.x + 0.5, bounds.origin.y + 0.5, bounds.width - 1, bounds.height - 1))
+
+        // Fill the disk with the black background color
+        NSColor.black.setFill()
+        disk.fill()
+
+        // Fill the disk with the background color
+        diskBackgroundColor.draw(in: disk, relativeCenterPosition: .zero)
+        // Draw the gradient within the disk
+        drawDisk(diskBackgroundColor, diskStrokeColor)
+        // Draw the symbol on the button
         drawSymbol(symbolColor)
+        // Draw a dimming effect if the button is highlighted or hovered
         drawDimming(disk)
     }
 
+    /// Draws a dimming effect on the button when it's highlighted or hovered.
+    ///
+    /// - Parameters:
+    ///   - disk: The `NSBezierPath` representing the button's disk.
     private func drawDimming(_ disk: NSBezierPath) {
+        // Set the line width of the disk
         disk.lineWidth = 1
-        if (isHighlighted) {
-            NSColor.black.withAlphaComponent(0.5).setFill()
-            disk.fill()
-        } else if (isMouseOver) {
+        if isMouseOver {
```

Comment:
You removed the difference of being highlighted = 50% opacity, and being hovered by the mouse = 25% opacity. Now the mouse hover is 25%, and the highlight is gone; it's always 100%. I'm not sure why this change is made. How is it related to the stated goal of the MR?

Reply:
The button is currently an NSView, and when it needs to be displayed, it should show its color at 100% opacity. When the mouse hovers over it, it should display at 25% opacity. The NSButton has an isHighlighted state, but this state is no longer needed with the current setup.